.github/workflows/test.yml

```diff
@@ -5,8 +5,6 @@ on:
     branches:
       - main
   pull_request:
-    branches:
-      - main
```

Comment:
Yes, I intended.
With this line, the open PR runs on top of the main branch structure, ignoring what is being added in the PR.
If you observe, node 20 does not appear in the test matrix just below, it will only appear in the next PR because it will be in main.

With this change, every change from the next PR will actually look at the created branch.